### PR TITLE
[Creación del Modelo Libro] - Creation book model

### PR DESCRIPTION
**Pull request by AleFalcon**
## Summary
Creation book model, with the mapping

## Trello Card
https://trello.com/c/pEhqUIXb/14-crear-modelo-de-book

### CODE REVIEW

File: src/main/java/wolox/training/models/Book.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
+package wolox.training.models;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+
+
+@Entity
+public class Book {
```
Comment (facucompe) [inline src/main/java/wolox/training/models/Book.java:11]: JPA needs an empty constructor to instantiate the class

**Review by facucompe — CHANGES_REQUESTED** [review]

File: src/main/java/wolox/training/models/Book.java (hunk 1 above)
Comment (AleFalcon) [reply src/main/java/wolox/training/models/Book.java:11]: I thnik that this is resolved

File: src/main/java/wolox/training/models/Book.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
```
Comment (Gianni-Nicolas) [inline src/main/java/wolox/training/models/Book.java]: In Postgres when GenerationType.AUTO is used, it generates an incremental IDs shared between the schema tables. This happens only for this database engine. To generate independent IDs for each table use GenerationType.SEQUENCE.

```
@GeneratedValue (strategy = GenerationType.SEQUENCE, generator = "BOOK_SEQ")            
@SequenceGenerator(name = "BOOK_SEQ", sequenceName = "BOOK_SEQ")
```

File: src/main/java/wolox/training/models/Book.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    private long id;
+
+    @Column(nullable = false, unique = true)
+    private String genre;
+
+    @Column(nullable = false, unique = true)
```
Comment (Gianni-Nicolas) [inline src/main/java/wolox/training/models/Book.java]: Use the unique attribute when the attribute is really unique.
In this case only the isbn attribute is unique.

File: src/main/java/wolox/training/models/Book.java [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+    private Integer getPage() {
+        return page;
+    }
+
+    private String getIsbn() {
+        return isbn;
+    }
+
+    private void setId(long id) {
```
Comment (Gianni-Nicolas) [inline src/main/java/wolox/training/models/Book.java]: The IDs generated by the entity, thanks to the @GeneratedValue annotation, make it unnecessary to include the method "public void setId (Long id)".

File: src/main/java/wolox/training/TrainingApplication.java [hunk 5]
```diff
@@ -2,7 +2,9 @@
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
 
+@EnableJpaRepositories("com.baeldung.persistence.repo")
```
Comment (Gianni-Nicolas) [inline src/main/java/wolox/training/TrainingApplication.java]: I don't think the EnableJpaRepositories annotation is necessary

**Review by Gianni-Nicolas — COMMENTED** [review]
Great job, I leave you just a few comments

File: src/main/java/wolox/training/models/Book.java (hunk 2 above)
Comment (AleFalcon) [reply src/main/java/wolox/training/models/Book.java]: okay

File: src/main/java/wolox/training/models/Book.java (hunk 3 above)
Comment (AleFalcon) [reply src/main/java/wolox/training/models/Book.java]: it is true

File: src/main/java/wolox/training/models/Book.java (hunk 4 above)
Comment (AleFalcon) [reply src/main/java/wolox/training/models/Book.java]: okay

File: src/main/java/wolox/training/TrainingApplication.java (hunk 5 above)
Comment (AleFalcon) [reply src/main/java/wolox/training/TrainingApplication.java]: okay

**Review by facucompe — APPROVED** [review]

**Review by Gianni-Nicolas — APPROVED** [review]

**Review by Gianni-Nicolas — APPROVED** [review]

**Pull request merged**